src/crimson/os/cyan_store.cc

```diff
@@ -201,7 +201,7 @@ seastar::future<ceph::bufferlist> CyanStore::read(CollectionRef c,
 
 seastar::future<ceph::bufferptr> CyanStore::get_attr(CollectionRef c,
                                                      const ghobject_t& oid,
-                                                     std::string_view name)
+                                                     std::string_view name) const
```

Comment:
I'm not sure it makes sense generally speaking to mark read operations like list_objects and get_attr const.  It's true that these calls don't mutate the store, however, essentially any implementation will have internal state that does mutate (caches, etc) as a consequence of the call.  Marking those things mutable would seem to me to be not really valid.

Reply:
I considered `const` in the spot of caching and mutexes (the former in the case of the old `ObjectStore` interface). Both of them still look as a private implementation detail which shouldn't affect the interface. Personally I don't perceive `const` as a guarantee there will be no modification of the underlying memory; [it's rather about visibility of modifications at the public interface's level](https://stackoverflow.com/a/105061).

Also I'm not sure that in-RAM caching will be always needed. Surely, for spinners and even for quite fast SSDs it's beneficial. However, the smaller the gap between RAM and storage speeds becomes, the less profit it brings.

Caching looks like one of the use cases where `mutable` fits perfectly.